tests/lib/linter.js

```diff
@@ -2711,35 +2711,48 @@ describe("Linter", () => {
     describe("verify()", () => {
         describe("filenames", () => {
             it("should allow filename to be passed on options object", () => {
+                const filenameChecker = sandbox.spy(context => {
+                    assert.strictEqual(context.getFilename(), "foo.js");
+                    return {};
+                });
 
-                linter.verify("foo;", {}, { filename: "foo.js" });
-                const result = linter.getFilename();
```

Comment:
@not-an-aardvark Could we keep track of the functions like that, that we could possibly remove/make private in the next major release?

Reply:
Discussed in https://github.com/eslint/eslint/pull/9219